### DISCUSSION TITLE
Why does discord.py sometimes spam and use old code that was deleted and saved?

### DISCUSSION THREAD

**Discussion by sirquackiii** [Rapptz/discord.py#6717] [Q&A] [answered]
So, this is a constant issue that I have had with discord.py. Sometimes, randomly, even after you haven't changed any code, it starts spamming the same message, and using old code that you have written and saved. It is very, VERY annoying. One bot I made just started spamming messages despite nothing in a loop. But the outright weirdest thing about this is that it somehow grabs old code that you deleted and uses it, but then it randomly switches back to your new code! That is creepy. I have absolutely no idea what is happening here, and I hope this gets fixed soon.

Thanks!

**Comment by bandwiches** [comment]
This is highly unlikely.  What's more likely is that there is something tucked away somewhere causing this to happen that you may not be aware of.

**Reply by sirquackiii** [reply]
```
import discord
from discord.ext import commands
import json
import random

x = open("data.json", "r")
client = commands.Bot(command_prefix = "#")
data = json.load(x)

@client.command()
async def register(ctx, name):
	await ctx.send(data)
	return

client.run("token")
```

That's my code. So what's causing this?

Thanks :)

**Reply by bandwiches** [reply]
Great.  Now what's the issue you're seeing?  In more detail.

For starters, you should close your IO buffer.

```python
x = open("data.json", "r")
client = commands.Bot(command_prefix = "#")
data = json.load(x)
x.close()
```

**Reply by bandwiches** [reply]
Also, why do you `import random` and never use it?  Same applies to `async def register(ctx, name):` you never use the parameter `name` in your function.

**Reply by sirquackiii** [reply]
hmm still spamming. also, the random module is used and also the "name" parameter but im just testing the command right now.

**Comment by Rapptz** [comment, marked as answer]
discord.py is a library -- it only does what you tell it to do. If you wrote something to spam then it'll spam. If you keep spawning processes without closing them then the old code will still be alive. If you want to restart your bot make sure to fully close it, otherwise a process will continue using "old code" as you put it.

**Reply by sirquackiii** [reply]
Thank you so much! No wonder nobody has asked this question before. :)
